patches/server/1045-Conduit-API.patch

```diff
@@ -0,0 +1,35 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tamion <70228790+notTamion@users.noreply.github.com>
+Date: Sat, 27 Jan 2024 20:46:40 +0100
+Subject: [PATCH] Conduit API
+
+== AT ==
+public net.minecraft.world.level.block.entity.ConduitBlockEntity effectBlocks
+public net.minecraft.world.level.block.entity.ConduitBlockEntity destroyTarget
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftConduit.java b/src/main/java/org/bukkit/craftbukkit/block/CraftConduit.java
+index 29bcac10a7edf53015941e4c28c4f2d9a5a3db56..24a631a7bbb7e8340a334f1b407b8427768dd1c2 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftConduit.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftConduit.java
+@@ -18,4 +18,21 @@ public class CraftConduit extends CraftBlockEntityState<ConduitBlockEntity> impl
+     public CraftConduit copy() {
+         return new CraftConduit(this);
+     }
++
++    // Paper start - Conduit API
++    @Override
++    public boolean isActive() {
++        return this.getTileEntity().isActive();
++    }
++
++    @Override
++    public int getRange() {
++        return this.getTileEntity().effectBlocks.size() / 7 * 16;
```

Comment:
Can we have diff on change stuff here? These magic numbers are scary.

Reply:
what do you mean by diff? i mean thats just simple vanilla logic for conduits. for every 7 prisma blocks you get 16 blocks of range